webapp/src/components/systemConsole/teamFilter/teamFilter.tsx

```diff
@@ -53,26 +54,25 @@ function TeamFilter({id, setSaveNeeded, onChange, config}: CustomComponentProps)
                 });
 
                 setSelectedTeams(initialOptions);
+                setInitialSetting(id, optionsToConfig(initialOptions));
             }
         };
 
         task();
-    }, [config.PluginSettings.Plugins]);
+    }, [config.PluginSettings.Plugins, id, setInitialSetting]);
 
-    const customComponents: CustomComponentsDefinition = useMemo(() => (
-        {
-            Option: CustomOption,
-            MultiValueContainer: CustomMultiValueContainer,
-        }
-    ), []);
+    const customComponents: CustomComponentsDefinition = useMemo(() => ({
+        Option: CustomOption,
+        MultiValueContainer: CustomMultiValueContainer,
+    }), []);
+
+    const optionsToConfig = (teams: DropdownOption[]): TeamFilterConfig => ({
+        filteredTeamIDs: teams.map((option) => option.value),
+    });
```

Comment:
This doesn't depend on any prop or state, so it can be moved outside of the component.

Reply:
Done.